rental_pricelist/models/sale.py

```diff
@@ -277,8 +276,18 @@ def product_uom_change(self):
                             key
                         )
                         break
+        if self.number_of_time_unit:
+            self.product_uom_qty = self.rental_qty * self.number_of_time_unit
         return super(SaleOrderLine, self).product_uom_change()
 
+    @api.onchange(
+        "product_id", "price_unit", "product_uom", "product_uom_qty", "tax_id"
+    )
+    def _onchange_discount(self):
```

Comment:
It was already be done in function rental_qty_number_of_days_change()

Reply:
I've just installed module rental_pricelist and follow these steps I can not reproduce the error. if you change the price list, It doesn't reset the product_uom_qty. So the price keeps. Did you installed some other modules, that change the product_uom_qty?

Reply:
In rental quotations, if you select amount of time, amount of units and change pricelists it resets product_uom_qty

I can make a video if still not sure